### DISCUSSION TITLE
Expand on create

### DISCUSSION THREAD

**Discussion by joranmulderij** [pocketbase/pocketbase#1764] [Q&A] [answered]
Can you do a single request and create a base records and many child records?

**Comment by ganigeorgiev** [comment, marked as answer]
Bulk and transactional insertion are not supported via the web APIs at the moment (_if that's what you are asking_) and you'll have to send individual requests for each record you want to create.

If you want to expand the created base/main record with its relation in the returned response, you can pass the `expand` query parameter:

```js
const result = await pb.collection("example").create({ someRelField: "someId" }, { expand: "someRelField" })
```

**Reply by joranmulderij** [reply]
Yes I understand this but now `someRelField` has now value right? I want to create another record and put the new id in the `someRelField` field.

**Reply by ganigeorgiev** [reply]
I'm sorry, I'm not entirely sure that I understand what do you mean. I've updated the example to include the relation field in the create data.

Could you provide a code sample (pseudo code is also fine) what you are trying to do?

**Reply by joranmulderij** [reply]
I would like to be able to do this:
```
const result = await pb.collection("example").create({ videos: [{'url':'<link>'}, {'url':'<link>'}] }, { expand: "videos" })
```
That would create 3 records. One in the `example` collection and two in the `videos` collection.

**Reply by joranmulderij** [reply]
Or:
```
const result = await pb.collection("example").create({ expand:{ videos: [{'url':'<link>'}, {'url':'<link>'}]} }, { expand: "videos" })
```

**Reply by ganigeorgiev** [reply]
I understand. This is not possible at the moment (_this was partially discussed in https://github.com/pocketbase/pocketbase/discussions/1735_). You'll have to send 3 individual requests:

```js
const video1 = await pb.collection("videos").create({ url: "<link>" })
const video2 = await pb.collection("videos").create({ url: "<link>" })
const example = await pb.collection("example").create({ videos: [video1.id, video2.id] }, { expand: "videos" })
```

If you want to execute them in a transaction you'll have to [use PocketBase as framework](https://pocketbase.io/docs/use-as-framework/) - [register your own route](https://pocketbase.io/docs/router/#register-new-route) and wrap the [record create](https://pocketbase.io/docs/record-methods/#create-new-record-without-data-validations) calls inside [`app.Dao().RunInTransaction()`](https://pocketbase.io/docs/database/#transaction). _A recent example for `RunInTransaction` you can find also in https://github.com/pocketbase/pocketbase/discussions/1763._

**Reply by joranmulderij** [reply]
Alright thanks

**Reply by vmmello** [reply]
> Bulk and transactional insertion are not supported via the web APIs at the moment (_if that's what you are asking_) and you'll have to send individual requests for each record you want to create.

Asking 2 years later the same question. Is the answer above still correct?

**Reply by ganigeorgiev** [reply]
@vmmello No. See https://pocketbase.io/docs/api-records/#batch-createupdateupsertdelete-records.
If you need further help, please open a new Q&A discussion but try to elaborate a little more on your use case.
